### ISSUE TITLE
aws-autoscaling: provide service limit warnings

### ISSUE THREAD

**Issue by qvissak** [aws/aws-cdk#25300]
### Describe the feature

CDK should inform users of potentially (or actually) hitting account service limit quotas, since at build time it knows the settings that will fail at runtime after cloud formation tries to deploy.

### Use Case

Recently when I went to create an ASG for my service's largest production region, I ran into a series of account service limits, which I feel could have been preempted at build time by CDK.

For example, I decided to use a small instance type, namely C5.Large, but use a lot of them. So, I set my min capacity to a small number (400) to let ASG do its thing, and set the max number to 3,000. Upon ramping up traffic to my new service, after 1,000 instances the ASG started to fail to launch new instances, because I hit the max number of targets behind a target group limit (1,000).

Later, I updated my configuration and went to go re-deploy just to run into another limit. This time while using C5.2XLarge, I hit the max vCPU for my account. I had to get either both or one of the tickets I filed with AWS Support approved before being able to ramp up to handle my production load.

### Proposed Solution

At build time, ASG CDK code could simply print a warning that setting max instances to 1,000+ could be problematic without increasing account service limits first of max TG instances. It could also do math on how many vCPU I could possibly have at max based on my instance type and max instances settings in order to pre-empt ASG from failing to launch due to my account hitting the max vCPU limit.

Getting more advanced, to avoid API calls at build time, the library could be supplemented with another library that has a static list of all of these limits for all services and each CDK package can integrate with it to print warnings/errors to users based on their configuration settings.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.73.0

### Environment details (OS name and version, etc.)

AL2

**Comment by khushail** [comment]
Thanks @qvissak for reaching out with this feature request. This would be really helpful to users if implemented. I am marking this as P2 which means our core team won't be able to work on it immediately. 

So if you would like to contribute, team would be happy to review your PR. Sharing the [contributing guide](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-1-find-something-to-work-on) to get started.

**Comment by pahud** [comment]
Nice idea! As CDK only manage resources created by itself, it would be great if we develop a framework like [cdk-ng](https://github.com/cdklabs/cdk-nag) to calculate existing resource usage in the [Aspects](https://docs.aws.amazon.com/cdk/v2/guide/aspects.html). For example, the soft limit of VPC per account per region is 5, if I already have 5 existing VPCs, before I deploy my CDK app with `new ec2.Vpc();`, the framework should throw a warning due to the potential error from cloudformation. This probably would involves some SDK call to lookup specific resources and making this framework modular might be a good idea.

**Comment by qvissak** [comment]
Agreed - this should definitely be modular so that it can be used with any library, not just aws-autoscaling.
